web/src/components/game/goals/GoalManagement.tsx

```diff
@@ -35,8 +37,38 @@ export default function GoalManagement() {
     const { showDetails } = settings;
 
     const [goalUploadOpen, setGoalUploadOpen] = useState(false);
+    const [loading, setLoading] = useState(false); // State to manage loading during deletion
+    const [deleteConfirmationText, setDeleteConfirmationText] = useState("");
+    const settingsDialogRef = useRef<DialogRef>(null); // Settings dialog
+    const deleteDialogRef = useRef<DialogRef>(null); // Delete confirmation dialog
 
-    const dialogRef = useRef<DialogRef>(null);
+    // Function to handle deleting all goals
+    const deleteAllGoals = async () => {
+        if (deleteConfirmationText !== "DELETE") {
+            // Handle failure silently
+            return;
+        }
+    
+        try {
+            setLoading(true);
+            const response = await fetch(`/api/games/${slug}/delete-all-goals`, {
+                method: 'DELETE',
+            });
+    
+            if (!response.ok) {
+                // Handle failure silently
+                return;
+            }
+    
+            // Refresh the goal list if needed
+            mutateGoals(); 
```

Comment:
We should probably call `mutateGoals` in the finally block in order to catch corner cases where we get partial deletion before running into a database error or similar